### DISCUSSION TITLE
Assigning tensorial material property through csv

### DISCUSSION THREAD

**Discussion by YuanxinXiao** [idaholab/moose#26373] [Q&A Modules: Solid mechanics]
Hi 

I want to read 5x5 tensor for my material property that I am reading it through the csv file, I am able to read to read it through the csv file but face difficulty in assigning it to material property

```
Mat_prop.csv

x1, x2, x3, x4, x5
1,10, 3, 5,6
5,20, 2, 8,6
2,1, 3, 9,1
1,2,3,4,5
3, 4,5,67
```

This material property will remain constant for whole simulation

For reading, I am using this
```
  [Samplers]
    [mat_prop]
    type = CSVSampler
    samples_file = 'E.csv'
    column_names = ‘x1 x2 x3 x4 x5’
    execute_on = 'initial timestep_end'
    []
  []

  [VectorPostprocessors]
    [mat_data]
    type = SamplerData
    sampler = mat_prop
    execute_on = 'initial timestep_end'
    []
  []
```

For material assignment, I am facing difficulty in assigning this 5x5 tensor to lambda????
```
  [Materials]
    [compute_stress]
      type = ComputeNeoHookeanStress
      lambda = ?????
      mu = 520
    []
```

**Comment by ngrilli** [comment]
@YuanxinXiao I am not sure if there is ready code in MOOSE for that.
I have developed this to read in 3x3 matrices for each element to initialize a tensor
representing plastic deformation gradient in my case:

https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/ComputeDislocationCrystalPlasticityStress.C

See the this part in initialization:

    for (unsigned int i = 0; i < 3; ++i) {
	  for (unsigned int j = 0; j < 3; ++j) {
        initial_Fp(i,j) = _read_initial_Fp->getData(_current_elem, 3*i+j);
	  }
	}

In my case the file is constituted by 9 rows and one column for each element.

**Reply by YuanxinXiao** [reply]
I got your point, is it possible for you to tell how you are reading csv in input file

**Reply by lindsayad** [reply]
This question is very similar to that posed in https://github.com/idaholab/moose/discussions/26369

**Reply by GiudGiud** [reply]
@mvkpenn22 that s the kind of code I m thinking for your problem of filling the large tensor as well.

**Reply by GiudGiud** [reply]
However I would not necessarily use the CSVReader, the PropertyFileRead can do the job and will let you do more types of spatially-dependent properties
